File: src/main/java/com/wanderers/hotelier_webservices/server/exception/CustomerDaoException.java
package com.wanderers.hotelier_webservices.server.exception;

import com.wanderers.hotelier_webservices.server.dao.impl.CustomerDaoImpl;

/**
 * Custom Exception class which used to throw failures in {@link CustomerDaoImpl}
 */
public class CustomerDaoException extends ServerException {

    private static final long serialVersionUID = 1L;

    /**
     * Class constructor
     *
     * @param message String value of Exception
     */
    public CustomerDaoException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: src/main/kotlin/com/wanderers/hotelier_webservices/server/exception/CustomerDaoException.kt
package com.wanderers.hotelier_webservices.server.exception

import com.wanderers.hotelier_webservices.server.dao.impl.CustomerDaoImpl

/**
 * Custom Exception class which used to throw failures in [CustomerDaoImpl]
 */
class CustomerDaoException
/**
 * Class constructor
 *
 * @param message String value of Exception
 */
    (message: String?, cause: Throwable?) :
    ServerException(message, cause) {
    companion object {
        private const val serialVersionUID = 1L
    }
}